### DISCUSSION TITLE
Model button pressed with different durations than stimuli

### DISCUSSION THREAD

**Discussion by alireza-kr** [cvnlab/GLMsingle#136] [Q&A] [answered]
I have two experiments. In the first experiment, each trial was presented for 6 seconds, followed by a button press by the participants. I am wondering how I can model the button press, considering that it is very fast and the stimulus duration (stimdur) must remain constant in GLMsingle. In the second experiment, each trial was presented for a shorter duration of 1.5 seconds, followed by a button press.

Could you please guide me on how I can model the button presses in these two experiments using GLMsingle?

I should add that the beta values for pressing the button are not important to me. I am simply modeling them in GLM.

**Comment by kendrickkay** [comment]
Hi Alireza - 

This is primarily in the domain of how the experimenter (you) wishes to characterize button-press-related BOLD signals.  In other words, it's sort of a modeling choice.

One approach is to just imagine that any button press related BOLD responses will just be "lumped into" the BOLD response associated with the trial.

Or, were you trying to characterize button press related responses separately from the main trial?

Note that within a range, even though button presses are brief, the associated (slow/delayed) BOLD response might be quite comparable to other short durations.  For example, the shape of the expected BOLD response to a 0.1-s square wave is similar to that from a 1-s square wave (assuming linear impulse response, etc.)

Another approach is to just model all instances of button presses (across all trials) as instances of a single condition.  That might work, depending on how you set up the rest of the design matrix.

Kendrick

**Reply by alireza-kr** [reply]
Hello

Thank you so much, Kendrick.

I want to model button presses separately from the main trials. I plan to model all instances of button presses (across all trials) as instances of a single condition, as you suggested. However, in GLMsingle, all trials must have the same duration. Since my main trials each have a duration of 6 seconds, I would need to set the duration for button presses to 6 seconds as well, which I don't think is acceptable.

**Comment by kendrickkay** [comment, marked as answer]
I see.  The question, I guess, is what do you think the response to your 6 second events will be like?  Do you think the BOLD response will be "extended" for 6 seconds?  If so, I agree that's hard to reconcile with the button presses.  One possible approach would be to use a stimulus duration of, say, 2 seconds.  Then you could model each condition with three "mini conditions" (for the first 2 seconds, second 2 seconds, and the third 2 seconds), and then you could create your button press condition (using the 2 second duration).    

Alternatively, you could just stick with the 6 seconds stimulus duration, and just use that to model your button presses.  There will likely be a timecourse shape mismatch for the button presses, but it may not be a huge problem, especially if you dont' really fundamentally care about the responses to the button presses.  (It will just be "time series modeling noise")

**Reply by alireza-kr** [reply]
Thanks, Kendrick.
